src/axom/quest/InOutOctree.hpp

```diff
@@ -60,3362 +65,1514 @@ namespace quest
+  /// against this segment's normal indicates queryPt's containment.
+  /// It is inside when the dot product is positive.
 
-    const VertexIndex numVertices = m_octree.m_meshWrapper.numMeshVertices();
-    for(VertexIndex i = 0; i < numVertices; ++i)
-    {
-      const SpacePt& pos = m_octree.m_meshWrapper.vertexPosition(i);
-      BlockIndex vertBlock = m_octree.findLeafBlock(pos);
-      const InOutBlockData& leafData = m_octree[vertBlock];
+  SLIC_ASSERT(leafData.color() == InOutBlockData::Gray);
+  SLIC_ASSERT(leafData.hasData());
 
-      VertexIndex vertInBlock = m_octree.leafVertex(vertBlock, leafData);
-      AXOM_DEBUG_VAR(vertInBlock);
+  GeometricBoundingBox blockBB = this->blockBoundingBox(leafBlk);
+  GeometricBoundingBox expandedBB = blockBB;
+  expandedBB.scale(m_boundingBoxScaleFactor);
 
-      // Check that we can find the leaf block indexing each vertex from its position
-      SLIC_ASSERT_MSG(
-        leafData.hasData() && vertInBlock == i,
-        fmt::format(
-          " Vertex {} at position {} was not indexed in block {} with bounding "
-          "box {} (point {} contained in block)."
-          " \n\n *** \n Leaf data: {} \n ***",
-          i,
-          pos,
-          vertBlock,
-          m_octree.blockBoundingBox(vertBlock),
-          (m_octree.blockBoundingBox(vertBlock).contains(pos) ? "is" : "is NOT"),
-          leafData));
-
-      // Check that our cached value of the vertex's block is accurate
-      SLIC_ASSERT_MSG(
-        vertBlock == m_octree.m_vertexToBlockMap[i],
-        fmt::format(
-          "Cached block for vertex {} differs from its found block. "
-          "\n\t -- cached block {} -- is leaf? {}"
-          "\n\t -- actual block {} -- is leaf? {}"
-          "\n\t -- vertex set size: {}",
-          i,
-          m_octree.m_vertexToBlockMap[i],
-          (m_octree[m_octree.m_vertexToBlockMap[i]].isLeaf() ? "yes" : "no"),
-          vertBlock,
-          (m_octree[vertBlock].isLeaf() ? "yes" : "no"),
-          numVertices));
-    }
-  }
+  SpacePt segmentPt;
 
-  void checkTrianglesReferencedInBoundaryVertexBlocks() const
+  CellIndexSet segmentSet = leafCells(leafBlk, leafData);
+  const int numSegments = segmentSet.size();
+  for(int i = 0; i < numSegments; ++i)
   {
-    SLIC_DEBUG(
-      "--Checking that each triangle is referenced by the leaf blocks "
-      "containing its vertices.");
+    /// Get the segment
+    CellIndex idx = segmentSet[i];
+    SpaceCell seg = m_meshWrapper.cellPositions(idx);
 
-    const axom::IndexType numTriangles = m_octree.m_meshWrapper.numMeshElements();
-    for(axom::IndexType tIdx = 0; tIdx < numTriangles; ++tIdx)
+    /// Find a point from this segment within the expanded bounding box of the mesh
+    // We'll use the midpoint of the segment after clipping it against the bounding box
+    double pMin, pMax;
+    const bool intersects = primal::intersect(seg, expandedBB, pMin, pMax);
```

Comment:
This uses a new formulation of `intersect(Segment, BoundingBox)` that returns the range of parametric coordinates of the Segment within the BoundingBox